### DISCUSSION TITLE
importing python in an app

### DISCUSSION THREAD

**Discussion by ely-peddler** [shorepine/tulipcc#398] [General]
I'm trying to split my app into multiple py files but I can't find a way to import the extra code 

I have a set up something like this
```
my_app/
    __init__.py
    my_app.py
    util.py
```

I've tried multiple import statements in my_app.py to try to include the code in util.py but they all fail for various reasons including AttributeError: 'module' object has no attribute '...' and ImportError: can't perform relative import

Is it possible to split a tulip app into multiple py files? If so how?

**Comment by bwhitman** [comment]
This should work! Are you using `run('my_app')` to start this app?

**Reply by ely-peddler** [reply]
yes

**Comment by bwhitman** [comment]
I just made a test case for this, seem to work for me. Folder `test` looks like

```
test/
    test.py
    a.py
```

`test.py` is:
```python
import a
print(a.a())
```

`a.py` is:
```python
def a():
    return "hello"
```

and then

```python
>>> run('test')
hello
```

What is different about what you're trying?

**Reply by ely-peddler** [reply]
might be that i'm importing classes - i will try your code and see if i get that working

**Reply by bwhitman** [reply]
A thing that may trip you up is if Tulip stays in the app directory when you go to the REPL. I think this happens if the app has an error. So double check when you're `run('my_app')` that you're in the directory that has your `my_app` directory in it.

**Reply by ely-peddler** [reply]
will do that, tried you example and extended it in various ways and they all work, so must be something i'm doing.

also it looks like i don't need a __init__.py, is that right?

**Reply by ely-peddler** [reply]
it's working now - i did nothing special - who knows what happened - sorry i know that's not that helpful but it's working so I'll stop bothering you at least

**Reply by bwhitman** [reply]
no bother! great you got it working!!
